pkg/build/builder/daemonless.go

```diff
@@ -311,118 +328,233 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 		PullPushRetryDelay:      DefaultPushOrPullRetryDelay,
 	}
 
-	_, _, err := imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
+	_, _, err = imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
 	return err
 }
 
-func generateTransientMounts() []string {
-	mounts := []string{}
-	mounts = appendRHSMMount(defaultMountStart, mounts)
-	mounts = appendETCPKIMount(defaultMountStart, mounts)
-	mounts = appendRHRepoMount(defaultMountStart, mounts)
-	mounts = appendCATrustMount(mounts)
+type TransientMount struct {
+	buildapiv1.BuildVolumeMount
+	MountPathSource string
+	// additional buildah mount options not exposed to the Build API
+	NoDev  bool
+	NoExec bool
+	NoSuid bool
+}
+
+func (tm TransientMount) String() string {
+	var mountOptions []string
+	if !tm.BuildVolumeMount.ReadOnly {
+		mountOptions = append(mountOptions, "rw")
+	}
+	if tm.NoDev {
+		mountOptions = append(mountOptions, "nodev")
+	}
+	if tm.NoExec {
+		mountOptions = append(mountOptions, "noexec")
+	}
+	if tm.NoSuid {
+		mountOptions = append(mountOptions, "nosuid")
+	}
+	return fmt.Sprintf("%s:%s:%s", tm.MountPathSource, tm.MountPath, strings.Join(mountOptions, ","))
```

Comment:
See my note about `MountPath` being relative - we should convert this to an absolute path.